### ISSUE TITLE
pip-compile doesn't clean up its temporary directories, generates warnings with PYTHONDEVMODE=1

### ISSUE THREAD

**Issue by si14** [jazzband/pip-tools#1209]
When run with `PYTHONDEVMODE=1` `pip-compile` emits a bunch of warnings like this:

```
/usr/local/lib/python3.8/tempfile.py:819: ResourceWarning: Implicitly cleaning up <TemporaryDirectory '/tmp/tmpfrhhtu6jbuild'>
  _warnings.warn(warn_message, ResourceWarning)
/usr/local/lib/python3.8/tempfile.py:819: ResourceWarning: Implicitly cleaning up <TemporaryDirectory '/tmp/tmp0o5twtbvsource'>
```

Quick skim through the sources reveals that temporary directories are created without explicit cleanup. Probably the easiest way to fix it is to use them as context managers.

#### Environment Versions

1. OS Type: any
1. Python version: 3.8
1. pip version: 20.0.2
1. pip-tools version: 5.3.1

#### Steps to replicate

1. run `PYTHONDEVMODE=1 pip-compile ...` on a nontrivial list of dependencies
2. see the warnings

#### Expected result

No warnings

#### Actual result

A bunch of warnings

**Comment by atugushev** [comment]
Hello @si14,

Thanks for the issue. It's fixed in #1233.

**Comment by si14** [comment]
Thanks a lot!
